gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/ExecutionTreeBuilder.java

```diff
@@ -147,13 +155,19 @@ public QueryNode build() {
      * {@link FilterOptimizingVisitor} is needed to merge filters corresponding to the same source
      * into one {@link DataFetcherNode}, instead of having multiple {@link DataFetcherNode}s for
      * each filter
+     *
+     * <p>It is not needed for AND filter, since the filter tree is already optimised with a single
+     * data fetcher node for each source
      */
-    QueryNode optimizedFilterTree = filterTree.acceptVisitor(new FilterOptimizingVisitor());
-    if (LOG.isDebugEnabled()) {
-      LOG.debug("Optimized Filter Tree:{}", optimizedFilterTree.acceptVisitor(new PrintVisitor()));
+    boolean isAndFilter = executionContext.getExpressionContext().isAndFilter();
```

Comment:
Didnt understand this. The comment says optimization visitor is not needed for AND filter but the condition is checking for isAndFilter. I'm confused.

Reply:
Thanks for catching it. I have fixed it. It should have been `!isAndFilter`